src/IO/S3/Credentials.cpp

```diff
@@ -717,8 +722,13 @@ S3CredentialsProviderChain::S3CredentialsProviderChain(
     AddProvider(std::make_shared<Aws::Auth::ProfileConfigFileAWSCredentialsProvider>());
 }
 
+std::string determineAvailabilityZone(const DB::S3::PocoHTTPClientConfiguration & configuration)
+{
+    auto aws_client_configuration = getAwsClientConfig(configuration);
+    auto ec2_metadata_client = InitEC2MetadataClient(aws_client_configuration);
+    return ec2_metadata_client->getCurrentAvailabilityZone();
```

Comment:
I think it will work perfectly for AWS, but interesting how it will work for GCP? Let's test. 

Maybe we need to support manual setting of the availability zone in the configuration file?

Reply:
Yes. GCP has metadata server available as well.

```sh
# On GKE cluster.
kubectl run testing-pod --image=curlimages/curl --command -- sleep 3600
kex testing-pod --  curl -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/instance/zone
projects/<redacted>/zones/us-east1-c
```

Reply:
Added GCP impl.